logai/algorithms/factory.py

```diff
@@ -58,14 +64,25 @@ def unregister(cls, task, name):
         """
         return cls._algorithms[task].pop(name, None)
 
+    def _check_algorithm(self, task, name):
+        if name in self._algorithms_with_torch:
+            if not is_torch_available():
+                raise ImportError("Python lib torch is not installed, "
+                                  "please install it via `pip install torch`.")
```

Comment:
This `ImportError` message is fine but we may not need to go into this detail. We simply combine all checks in on condition and if any the checked package is not available here, raise `ImportError` with message like "please install the missing packages via `pip install logai[deep-learning]`. Also easier for version control.